src/posix/platform/resolver.cpp

```diff
@@ -129,15 +141,100 @@ void Resolver::LoadDnsServerListFromConf(void)
+{
+    otBorderRoutingPrefixTableIterator iterator;
+    otBorderRoutingRdnssAddrEntry      entry;
+    otBorderRoutingRdnssAddrEntry      rdnssEntries[kMaxRecursiveServerCount + 1];
+    otIp6Address                       rdnssServers[kMaxRecursiveServerCount];
+    uint32_t                           numEntries = 0;
+
+    otBorderRoutingPrefixTableInitIterator(gInstance, &iterator);
+
+    while (otBorderRoutingGetNextRdnssAddrEntry(gInstance, &iterator, &entry) == OT_ERROR_NONE)
+    {
+        uint32_t i = 0;
+
+        // Check if the entry address is already in the list.
+        for (; i < numEntries; ++i)
+        {
+            if (otIp6IsAddressEqual(&entry.mAddress, &rdnssEntries[i].mAddress))
+            {
+                rdnssEntries[i].mLifetime = OT_MAX(rdnssEntries[i].mLifetime, entry.mLifetime);
+
+                break;
+            }
+        }
+
+        // If the address is not a duplicate, add the entry to the entry list.
+        if (i == numEntries)
+        {
+            rdnssEntries[numEntries++] = entry;
+
+            std::sort(rdnssEntries, rdnssEntries + numEntries,
+                      [](const otBorderRoutingRdnssAddrEntry &a, const otBorderRoutingRdnssAddrEntry &b) {
+                          return a.mLifetime > b.mLifetime;
+                      });
+
+            numEntries = OT_MIN(numEntries, kMaxRecursiveServerCount);
+        }
+    }
+
+    for (uint32_t i = 0; i < numEntries; i++)
+    {
+        rdnssServers[i] = rdnssEntries[i].mAddress;
+    }
+
+    SetRecursiveDnsServerList(rdnssServers, numEntries);
+}
+#endif // OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
+
+otError Resolver::SendQueryToServer(Transaction        *aTxn,
+                                    const otIp6Address &aServerAddress,
+                                    const char         *aPacket,
+                                    uint16_t            aLength)
 {
-    char         packet[kMaxDnsMessageSize];
-    otError      error  = OT_ERROR_NONE;
-    uint16_t     length = otMessageGetLength(aQuery);
     otIp4Address ip4Addr;
     sockaddr_in  serverAddr4;
     sockaddr_in6 serverAddr6;
 
+    if (otIp4FromIp4MappedIp6Address(&aServerAddress, &ip4Addr) == OT_ERROR_NONE)
+    {
+        memcpy(&serverAddr4.sin_addr.s_addr, &ip4Addr, sizeof(otIp4Address));
+        serverAddr4.sin_family = AF_INET;
+        serverAddr4.sin_port   = htons(53);
+
+        VerifyOrExit(sendto(aTxn->mUdpFd4, aPacket, aLength, MSG_DONTWAIT, reinterpret_cast<sockaddr *>(&serverAddr4),
+                            sizeof(serverAddr4)) > 0);
+    }
+    else
+    {
+        memcpy(&serverAddr6.sin6_addr, &aServerAddress, sizeof(otIp6Address));
+        serverAddr6.sin6_family = AF_INET6;
+        serverAddr6.sin6_port   = htons(53);
+
+        VerifyOrExit(sendto(aTxn->mUdpFd6, aPacket, aLength, MSG_DONTWAIT, reinterpret_cast<sockaddr *>(&serverAddr6),
+                            sizeof(serverAddr6)) > 0);
+    }
+
+    return OT_ERROR_NONE;
```

Comment:
We should have a single `return` from any method/functon. Use `otError error` and set it.

Per style guie:
>   - There should be one return statement per free function or method at the end of the free function or method.